File: src/main/java/no/novari/flyt/configuration/validation/constraints/IntegrationAndMetadataMatches.java
package no.novari.flyt.configuration.validation.constraints;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = IntegrationAndMetadataMatchesValidator.class)
public @interface IntegrationAndMetadataMatches {

    String message() default "Integration and metadata do not have matching source application ids and/or source application integration ids";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/constraints/IntegrationAndMetadataMatches.kt
package no.novari.flyt.configuration.validation.constraints

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Constraint(validatedBy = [IntegrationAndMetadataMatchesValidator::class])
annotation class IntegrationAndMetadataMatches(
    val message: String =
        "Integration and metadata do not have matching source application ids and/or source application integration ids",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
